src/ui/public/vislib/lib/axis_scale.js

```diff
@@ -184,58 +142,37 @@ export default function AxisScaleFactory(Private) {
       throw new errors.InvalidLogScaleValues();
     };
 
-    /**
-     * Return the domain for log scale, i.e. the extent of the log scale.
-     * Log scales must begin at 1 since the log(0) = -Infinity
-     *
-     * @param scale
-     * @param yMin
-     * @param yMax
-     * @returns {*[]}
-     */
     logDomain(min, max) {
       if (min < 0 || max < 0) return this.throwLogScaleValuesError();
       return [1, max];
     };
 
-    /**
-     * Returns the appropriate D3 scale
-     *
-     * @param fnName {String} D3 scale
-     * @returns {*}
-     */
     getScaleType(fnName) {
       if (fnName === 'square root') fnName = 'sqrt'; // Rename 'square root' to 'sqrt'
       fnName = fnName || 'linear';
 
-      if (this.isTimeDomain()) return d3.time.scale.utc(); // allow time scale
-      if (this.isOrdinal()) return d3.scale.ordinal();
+      if (this.config.isTimeDomain()) return d3.time.scale.utc(); // allow time scale
+      if (this.config.isOrdinal()) return d3.scale.ordinal();
       if (typeof d3.scale[fnName] !== 'function') return this.throwCustomError('Axis.getScaleType: ' + fnName + ' is not a function');
 
       return d3.scale[fnName]();
     };
 
-    /**
-     * Creates the d3 y scale function
-     *
-     * @method getscale
-     * @param length {Number} DOM Element height
-     * @returns {D3.Scale.QuantitiveScale|*} D3 scale function
-     */
     getScale(length) {
-      const scale = this.getScaleType(this.attr.scale);
+      const scale = this.getScaleType(this.config.get('vis._attr.scale'));
       const domain = this.getExtents();
       const range = this.getRange(length);
       this.scale = scale.domain(domain);
-      if (this.isOrdinal()) {
+      if (this.config.isOrdinal()) {
         this.scale.rangeBands(range, 0.1);
       } else {
         this.scale.range(range);
       }
 
-      if (!this.isUserDefined() && !this.isOrdinal() && !this.isTimeDomain()) this.scale.nice(); // round extents when not user defined
+      if (!this.isUserDefined() && !this.config.isOrdinal() && !this.config.isTimeDomain()) this.scale.nice(); // round extents when not user defined
       // Prevents bars from going off the chart when the y extents are within the domain range
-      if (this.axis._attr.type === 'histogram' && this.scale.clamp) this.scale.clamp(true);
+      // todo: make this work (its accessing chart config ... we only have axis config availible)
+      //if (this.axis._attr.type === 'histogram' && this.scale.clamp) this.scale.clamp(true);
```

Comment:
in case of histogram chart we want to make sure that the last column is the same size as the previous ones ...